### PR TITLE
Add renovate GitHub action

### PR DESCRIPTION
**Pull request by raynhamdev**
# Background

As part of [Sensible Default](https://docs.google.com/document/d/13RkZtQc4wgyUAcdzhVG9XCl-qJJBZazqA4ZZG0ado0U/edit#heading=h.5l19v0q8dv5o) work, we'd like to add renovate to enable automatically dependency update in this repository.

Both Renovate and Dependabot are working well with GitHub. We chose Renovate because it is easier to configure for grouped PR than Dependabot. Additionally, Renovate has been used in many Octopus repositories (including Server) and has proved it is integrated well with OctoBob to approve/merge dependency update PRs automatically.

This PR adds Renovate GitHub action to the repo.
- Run at 8AM, Mon-Fri
- Auto-merge is not yet enabled.
- Auto-approval by OctoBob is enabled.

I haven't enabled auto merge yet because I'd like to observe how integration with OctoBob works first. This is the part I was unable to test with my forked repository. Additional PR will be added later to enable auto-merge when we confirm this is working fine.

[sc-63579]

# Results

Tested from a forked repository.
image

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

**Review by sburmanoctopus — COMMENTED** [review]
The file `apprrove-renovate-pull-request.yml` looks like a typo. I.e. 2 Rs in `apprrove`. 

FYI, I'm not reviewing this PR, as I don't know how this stuff works. I was just looking and saw this 😁

File: renovate-config.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    // Renovate will create a new issue in the repository.
+    // This issue has a "dashboard" where you can get an overview of the status of all updates.
+    // https://docs.renovatebot.com/key-concepts/dashboard/
+    dependencyDashboard: true,
+    dependencyDashboardTitle: "Tentacle Dependency Dashboard",
+
+    platform: "github",
+    repositories: ["OctopusDeploy/OctopusTentacle"],
+    reviewers: ["raynhamdev"],
```
Comment (evolutionise) [inline renovate-config.js:31]: Let's make this the SaST GitHub team instead

File: renovate-config.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    labels: ["dependencies", "Tentacle"],
+    branchPrefix: "renovate-dotnet/",
+
+    // Limit the amount of PRs created
+    prConcurrentLimit: 10,
```
Comment (evolutionise) [inline renovate-config.js:36]: This feels really high, shall we set it to something like 2 to start? I don't want us to have a lot of them open and feel pressured to merge a bunch at once, or get overwhelmed by the volume.

File: renovate-config.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    prHourlyLimit: 2,
+
+    // If set to false, Renovate will upgrade dependencies to their latest release only. Renovate will not separate major or minor branches.
+    // https://docs.renovatebot.com/configuration-options/#separatemajorminor
+    separateMajorMinor: false,
```
Comment (evolutionise) [inline renovate-config.js:41]: This one is interesting. Other than .NET which is excluded above, are there any dependencies where we'd want to keep on an LTS version specifically? 
The docs do recommend setting this to `true` so I'm interested in why we went this way.

**Review by evolutionise — COMMENTED** [review]
A couple of minor comments, the only change I'd really like to see is the reviewers being the whole team rather than you specifically

File: renovate-config.js (hunk 3 above)
Comment (raynhamdev) [reply renovate-config.js:41]: I don't notice we have any dependencies that require us to stick to a LTS version. I set this to false on purpose to reduce the noise/number of PRs that are automatically created.

**Review by evolutionise — APPROVED** [review]
LGTM - nice work Ray, this was well-thought out and you did a great job of understanding the specific requirements for Tentacle vs other repos.

**Pull request merged**